tools/make/packaging.mk

```diff
@@ -31,63 +31,65 @@ dist-agent-binaries: dist/grafana-agent-linux-amd64       \
                      dist/grafana-agent-linux-amd64-boringcrypto  \
                      dist/grafana-agent-linux-arm64-boringcrypto
 
-dist/grafana-agent-linux-amd64: GO_TAGS += builtinassets promtail_journal_enabled
+dist/grafana-agent-linux-amd64: GO_TAGS += netgo builtinassets promtail_journal_enabled
 dist/grafana-agent-linux-amd64: GOOS    := linux
 dist/grafana-agent-linux-amd64: GOARCH  := amd64
 dist/grafana-agent-linux-amd64: generate-ui
 	$(PACKAGING_VARS) AGENT_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agent
 
-dist/grafana-agent-linux-arm64: GO_TAGS += builtinassets promtail_journal_enabled
+dist/grafana-agent-linux-arm64: GO_TAGS += netgo builtinassets promtail_journal_enabled
 dist/grafana-agent-linux-arm64: GOOS    := linux
 dist/grafana-agent-linux-arm64: GOARCH  := arm64
 dist/grafana-agent-linux-arm64: generate-ui
 	$(PACKAGING_VARS) AGENT_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agent
 
-dist/grafana-agent-linux-ppc64le: GO_TAGS += builtinassets promtail_journal_enabled
+dist/grafana-agent-linux-ppc64le: GO_TAGS += netgo builtinassets promtail_journal_enabled
 dist/grafana-agent-linux-ppc64le: GOOS    := linux
 dist/grafana-agent-linux-ppc64le: GOARCH  := ppc64le
 dist/grafana-agent-linux-ppc64le: generate-ui
 	$(PACKAGING_VARS) AGENT_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agent
 
-dist/grafana-agent-linux-s390x: GO_TAGS += builtinassets promtail_journal_enabled
+dist/grafana-agent-linux-s390x: GO_TAGS += netgo builtinassets promtail_journal_enabled
 dist/grafana-agent-linux-s390x: GOOS    := linux
 dist/grafana-agent-linux-s390x: GOARCH  := s390x
 dist/grafana-agent-linux-s390x: generate-ui
 	$(PACKAGING_VARS) AGENT_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agent
 
-dist/grafana-agent-darwin-amd64: GO_TAGS += builtinassets
+dist/grafana-agent-darwin-amd64: GO_TAGS += netgo builtinassets
 dist/grafana-agent-darwin-amd64: GOOS    := darwin
 dist/grafana-agent-darwin-amd64: GOARCH  := amd64
 dist/grafana-agent-darwin-amd64: generate-ui
 	$(PACKAGING_VARS) AGENT_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agent
 
-dist/grafana-agent-darwin-arm64: GO_TAGS += builtinassets
+dist/grafana-agent-darwin-arm64: GO_TAGS += netgo builtinassets
 dist/grafana-agent-darwin-arm64: GOOS    := darwin
 dist/grafana-agent-darwin-arm64: GOARCH  := arm64
 dist/grafana-agent-darwin-arm64: generate-ui
 	$(PACKAGING_VARS) AGENT_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agent
 
+# NOTE(rfratto): do not use netgo when building Windows binaries, which
+# prevents DNS short names from being resovable. See grafana/agent#4665.
```

Comment:
Do we need to add a TODO to revert this change when the underlying Go issue is resolved?
https://github.com/golang/go/issues/57757

Reply:
I don't think that's the same issue, is it? I don't see any mention of the hosts file in the issue or the Prometheus issue.

Reply:
Ah yes, I think you're right and it's not the same issue. But it's probably another issue which needs to be resolved before we use netgo on windows. Anyway, thank you for adding the TODO.